src/main/java/org/myrobotlab/service/Arduino.java

```diff
@@ -443,31 +448,69 @@ public void attachMotorControl(MotorControl motor) throws Exception {
 
   public void attachServoControl(ServoControl servo) throws Exception {
     if (isAttached(servo)) {
-      return; // already attached
+      return; 
     }
-    // query configuration out
+
     int pin = getAddress(servo.getPin());
     // targetOutput is ALWAYS ALWAYS degrees
     double targetOutput = servo.getTargetOutput();
-    double velocity = (servo.getSpeed() == null)?-1:servo.getSpeed();
-
-    // this saves original "attach" configuration - and maintains internal
-    // data
-    // structures
-    // and does DeviceControl.attach(this)
-    Integer deviceId = attachDevice(servo, new Object[] { pin, targetOutput, velocity });
+    double velocity = (servo.getSpeed() == null) ? -1 : servo.getSpeed();
 
-    // send data to micro-controller - convert degrees to microseconds
-    int uS = degreeToMicroseconds(targetOutput);
-    msg.servoAttach(deviceId, pin, uS, (int) velocity, servo.getName());
+    // add a device to our deviceList
+    DeviceMapping dm = attachDevice(servo, new Object[] { pin, targetOutput, velocity });
+    
+    if (isConnected()) {
+      int uS = degreeToMicroseconds(servo.getTargetOutput());
+      msg.servoAttach(dm.getId(), pin, uS, (int) velocity, servo.getName());
+      if (servo.isEnabled()) {
+        msg.servoAttachPin(dm.getId(), pin);
+      }
+    }
 
     servo.attach(this);
   }
 
+  /**
+   * 
+   * @param dm
+   */
+  public void reattach(DeviceMapping dm) {
+    Attachable attachable = dm.getDevice();

```

Comment:
This looks hard coded to only support reattaching servos and hobby servos?  or is this some sort of override because both Servo and HobbyServo reference the same MrlServo device.

It seems like any MrlDevice should be able to be re-attached.. not just Servo/HobbyServo

Reply:
agreed - although the details on reattach might depend on device. I've created a ticket for this #514